sdlib/dmd/thread.d

```diff
@@ -83,13 +89,32 @@ void __sd_gc_pre_suspend_hook(void* stackTop) {
 
 void __sd_gc_post_suspend_hook() {
 	thread_postSuspend();
+
+	import d.gc.tcache;
+	threadCache.stackTop = null;
 }
 
 void __sd_gc_pre_stop_the_world_hook(void* stackTop) {
 	thread_preStopTheWorld();
+
+	/**
+	 * For druntime, we need to determine if this thread is going to be
+	 * scanned by druntime. This means we have to call the suspend hook.
+	 * However, if this thread isn't being handled by druntime, we want to
+	 * defer to the normal thread scanning done in the mark function.
+	 * Therefore, the stack top should be either null or SkipScanStack.
+	 */
+	__sd_gc_pre_suspend_hook(stackTop);
```

Comment:
The stackTop here is not really meaningful (and in fact, you almost certainly expect that to be a managed thread and not use it at all). I don't think there is anything useful this function could ever do with the stack top, so there is no point adding it to the API.

You can get it from intrinsics or maybe even just pass null in there.